packages/nimble-components/src/wafer-map/index.ts

```diff
@@ -244,14 +266,65 @@ export class WaferMap extends FoundationElement {
                 this.dataManager.updateDiesRenderInfo();
                 this.renderer.updateSortedDiesAndDrawWafer();
             } else if (this.waferMapUpdateTracker.requiresDrawnWaferUpdate) {
-                this.renderer.drawWafer();
+                await this.renderer.drawWafer();
             }
+            await this.drawWafer();
             this.zoomHandler.connect();
         } else if (this.waferMapUpdateTracker.requiresRenderHoverUpdate) {
             this.renderer.renderHover();
         }
     }
 
+    private async drawWafer(): Promise<void> {
+        if (this.diesTable === undefined) {
+            return;
+        }
+        const waferMapMatrix = {
+            colIndexes: this.diesTable
+                .getChild('colIndex')
+                ?.toArray() as number[],
+            rowIndexes: this.diesTable
+                .getChild('rowIndex')
+                ?.toArray() as number[],
+            values: this.diesTable.getChild('value')?.toArray() as number[]
+        } as WaferMapMatrix;
+        await this.workerOne.updateMatrix(waferMapMatrix);
```

Comment:
There seem to be a bunch of `await` calls on methods being called on `workerOne`, but _none_ of them are actually `async` right?

Reply:
Did a bit of research, the Comlink wrapper makes the methods appear as asynchronous from the main thread perspective, though this doesn't make them to run asynchronously inside the worker, good catch! I'll make them async

Reply:
So, I think you may have been too hasty with the changes you've made. After talking with @rajsite, it's fine and expected that the worker method implementations need not be written as `async` if there is nothing fundamentally `async` about them (which there isn't), and it is also expected that you must await those methods exposed via the Comlink wrapper.

Reply:
Indeed, I'll revert the changes